src/history/hs.c

```diff
@@ -898,15 +898,19 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
                 continue;
             }
 
-            /* Skip updates that are already in the history store or are obsolete. */
-            if (F_ISSET(upd, WT_UPDATE_HS | WT_UPDATE_OBSOLETE)) {
+            /* Skip updates that are already in the history store. */
+            if (F_ISSET(upd, WT_UPDATE_HS)) {
                 if (hs_inserted)
                     WT_ERR_PANIC(session, WT_PANIC,
-                      "Inserting updates older than obsolete updates or updates that are already "
-                      "in the history store to the history store may corrupt the data.");
+                      "Inserting updates older than updates that are already in the history store "
+                      "to the history store may corrupt the data.");
```

Comment:
Update the comment to indicate that reinserting an update to history store could result in triggering out of order timestamped updates handling.